doc/cli_guidelines.md

```diff
@@ -0,0 +1,196 @@
+
+$ dinstaller config delete <list-key> [<key>=]<value> ...
+Deletes any entry matching all the given key-value pairs from a config list. The key is omitted for a list of scalar values.
+
+$ dinstaller config check
+Validates the config and prints errors
+
+$ dinstaller config info <key> [<value>]
+Prints info about the given key. If no value is given, then it prints what values are admitted by the given key. If a value is given, then it shows extra info about such a value.
+
+$ dinstaller summary [<section>]
+Prints a summary with the actions to perform in the system. If a section is given (e.g., storage, software, ...), then it only shows the section summary.
+
+$ dinstaller questions
+Prints questions and allows to answer them.
+
+~~~
+
+In those commands `<key>` represents a YAML key from the config structure and `<value>` is the value associated to the given key.
+
+### YAML Config
+
+The config settings are defined by this YAML structure:
+
+~~~
+---
+product: "Tumbleweed"
+
+languages:
+  - "es_ES"
+  - "en_US"
+
+user:
+  name: "test"
+  fullname: "User Test"
+  password: "n0ts3cr3t"
+  autologin: true
+
+root:
+  ssh_key: "1234abcd"
+  password: "n0ts3cr3t"
+
+storage:
+  candidate_devices:
+    - /dev/sda
+  lvm: true
+  encryption_password: 12345
+  volumes:
+    - mountpoint: /
+      fstype: btrfs
+    - mountpoint: /home
+      fstype: ext4
+      minsize: 10GiB
+~~~
+
+Nested keys are referenced in commands by joining them with dots (e.g., `user.name`).
+
+### Examples
+
+Let's see some examples:
+
+~~~
+# Set a product
+$ dinstaller config set product=Tumbleweed
+
+# Set user values
+$ dinstaller config set user.name=linux
+$ dinstaller config set user.fullname=linux
+$ dinstaller config set user.password=linux
+$ dinstaller config set user.name=linux user.fullname=linux user.password=12345
+
+# Unset user
+$ dinstaller config unset user
+
+# Add and delete languages
+$ dinstaller config add languages en_US
+$ dinstaller config delete languages en_US
+
+# Set storage settings
+$ dinstaller config set storage.lvm=false
+$ dinstaller config set storage.encryption_password=12345
+
+# Add and delete candidate devices
+$ dinstaller config add storage.candidate_devices /dev/sda
+$ dinstaller config delete storage.candidate_devices /dev/sdb
+
+# Add and delete storage volumes
+$ dinstaller config add storage.volumes mountpoint=/ minsize=10GiB
+$ dinstaller config delete storage.volumes mountpoint=/home
+
+# Reset storage config
+$ dinstaller config reset storage
+
+# Show some config values
+$ dinstaller config show storage.candidate_devices
+$ dinstaller config show user
+
+# Dump config into a file
+$ dinstaller config show > ~/config.yaml
+
+# Show info of a key
+$ dinstaller config info storage.candidate_devices
+$ dinstaller config info languages
+
+# Show info of a specific value
+$ dinstaller config info storage.candidate_devices /dev/sda
```

Comment:
The idea is to show extra info. For example, for a disk we can print its size, operating systems it contains, UUID, free size, etc. For a product, it could show the product description, etc.